client/cmd/job/change_namespace.go

```diff
@@ -118,25 +120,252 @@ func (c *changeNamespaceCommand) sendChangeNamespaceRequest(jobName string) erro
+	var parentsFilePathList []string
+	var errorMsg string
+	splitDirPaths := strings.Split(jobDirPath, "/")
+	for i := range splitDirPaths {
+		pathNearSpecIdx := len(splitDirPaths) - i
+		parentDirPath := filepath.Join(splitDirPaths[:pathNearSpecIdx]...)
+		parenFilePath := filepath.Join(parentDirPath, "this.yaml")
+		exist, err := afero.Exists(fileFS, parenFilePath)
+		if err != nil {
+			errorMsg += fmt.Sprintf("Unable to read directory %s to check 'this.yaml', err: %s\n", parenFilePath, err.Error())
+			continue
+		}
+		if exist {
+			parentsFilePathList = append(parentsFilePathList, parenFilePath)
+		}
+	}
+	var err error
+	if len(errorMsg) > 0 {
+		err = errors.New(errorMsg)
+	}
+	return parentsFilePathList, err
+}
+
+func mergeYAMLFiles(fs afero.Fs, sourcePath, targetPath string) error {
+	sourceObj, err := readYaml(fs, sourcePath)
+	if err != nil {
+		return err
+	}
+	targetObj, err := readYaml(fs, targetPath)
+	if err != nil {
+		return err
+	}
+	mergedObj, err := unionInterface(sourceObj, targetObj)
+	if err != nil {
+		return err
+	}
+	mergedFile, err := yaml.Marshal(mergedObj)
+	if err != nil {
+		return err
+	}
+	return afero.WriteFile(fs, targetPath, mergedFile, os.ModePerm)
+}
+
+func unionInterface(value1, value2 interface{}) (interface{}, error) {
+	val1 := reflect.ValueOf(value1)
+	val2 := reflect.ValueOf(value2)
+
+	if val1.Kind() != val2.Kind() {
+		return nil, errors.New("mismatched types")
+	}
+
+	switch val1.Kind() {
+	case reflect.Map:
+		merged := make(map[string]interface{})
+		iter1 := val1.MapRange()
+		for iter1.Next() {
+			key := iter1.Key().Interface()
+			value := iter1.Value().Interface()
+			merged[key.(string)] = value
+		}
+		iter2 := val2.MapRange()
+		for iter2.Next() {
+			key := iter2.Key().Interface()
+			value := iter2.Value().Interface()
+			if existingValue, exists := merged[key.(string)]; exists {
+				mergedValue, err := unionInterface(existingValue, value)
+				if err != nil {
+					return nil, err
+				}
+				merged[key.(string)] = mergedValue
+			} else {
+				merged[key.(string)] = value
+			}
+		}
+		return merged, nil
+
+	default:
+		return value2, nil
+	}
+}
+
+func findConflicts(incomming, existing interface{}, incomingFileName, existingFileName, path string) []string {
+	val1 := reflect.ValueOf(incomming)
+	val2 := reflect.ValueOf(existing)
+	var diffrences []string
+
+	if val1.Kind() != val2.Kind() {
+		diffrences = append(diffrences, fmt.Sprintf("Mismatched types at path %s\n", path))
+		return diffrences
+	}
+
+	switch val1.Kind() {
+	case reflect.Map:
+		iter := val1.MapRange()
+		for iter.Next() {
+			key := iter.Key()
+			val1Child := iter.Value()
+			val2Child := val2.MapIndex(key)
+			newPath := fmt.Sprintf("%s.%v", path, key.Interface())
+			if val2Child.IsValid() {
+				diffrences = append(diffrences, findConflicts(val1Child.Interface(), val2Child.Interface(), incomingFileName, existingFileName, newPath)...)
+			}
+		}
+
+	case reflect.Slice, reflect.Array:
+		len1 := val1.Len()
```

Comment:
I assume the ordering of the array should be sorted before. It won't work for 2 arrays with same element but different order